hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/record/ExternalizedRecordCustomizer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.workflows.record;
+
+import com.hedera.hapi.node.base.Transaction;
+import java.util.function.UnaryOperator;
+
+/**
+ * Allows a {@link com.hedera.node.app.spi.workflows.TransactionHandler} that dispatches child transactions
+ * to customize exactly how these records are externalized. Specifically, it allows the handler to,
+ * <ul>
+ *     <li>Completely suppress the record because it contains redundant information (as in the case of
+ *     the child transaction dispatched to implement a top-level HAPI {@code ContractCreate}).</li>
+ *     <li>Transform the dispatched {@link Transaction} immediately before it is externalized (as
+ *     in the case of the child {@link com.hedera.hapi.node.token.CryptoCreateTransactionBody} dispatched
+ *     to implement an internal contract creation, which should be externalized as an equivalent
+ *     {@link com.hedera.hapi.node.contract.ContractCreateTransactionBody}.</li>
```

Comment:
Looks like `apply` is never called if `shouldSuppressRecord` returns `true`. (I infer this from the two implementations where `apply` simply throws an exception.). Could this be explicitly mentioned here?

I really _like_ this kind of customization point, BTW.

(Though an alternative would be an `apply` that returned an `Optional`.  (not extending `UnaryOperator` of course.) Java pattern matching in switch statements is almost enough to help but the rest of the type system isn't strong enough to make the whole thing worth it yet.)

Reply:
Added a comment and [convenience implementation](https://github.com/hashgraph/hedera-services/blob/27c8a835a98208da9326948af21f0c7c7dfb08d3/hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/record/ExternalizedRecordCustomizer.java#L43) that suppresses the record and throws on receiving an `apply()` call.

Reply:
Yes! That's what I should have suggested ... well, next time ....